scripts/dev/utils/push_and_pr.py

```diff
@@ -0,0 +1,125 @@
+if PROJECT_ROOT not in sys.path:
+    sys.path.insert(0, PROJECT_ROOT)
+
+from scripts.dev.utils.git_repo import GitRepo
+
+def push_branch_and_create_pr(git_obj, merge_branch_name, work_item_id, username):
+    """
+    Push the specified branch to the fork remote and create a pull request.
+
+    :param git_obj: The GitRepo object to interact with the repository.
+    :param merge_branch_name: The name of the branch to push and create a PR from.
+    :param work_item_id: The work item ID to reference in the PR description.
+    :param username: The GitHub username owning the downstream fork.
+    :return: A tuple (status_code, message). Returns (0, None) on success.
+    """
+    push_details = push_branch(git_obj, merge_branch_name)
+    if push_details[0] != 0:
+        print(push_details[1])
+        return push_details
+
+    pr_details = create_pr(git_obj, merge_branch_name, work_item_id, username)
+    if pr_details[0] != 0:
+        print(pr_details[1])
+        return pr_details
+
+    return (0,None)
+
+def push_branch(git_obj, merge_branch_name):
+    """
+    Push the specified branch to the fork remote, deleting it first if it already exists.
+
+    :param git_obj: The GitRepo object to interact with the repository.
+    :param merge_branch_name: The name of the branch to push.
+    :return: A tuple (status_code, message). Returns (0, None) on success.
+    """
+    add_remote_details = git_obj.add_remote(git_obj.fork_name, git_obj.fork_url)
+    if add_remote_details[0] != 0:
+        print(add_remote_details[1])
+        return add_remote_details
+
+    if git_obj.branch_exists(merge_branch_name, git_obj.fork_name, check_on_remote=True):
+        push_delete_details = git_obj.push(merge_branch_name, git_obj.fork_name, delete=True)
+        if push_delete_details[0] != 0:
+            print(push_delete_details[1])
+            return push_delete_details
+
+    push_details = git_obj.push(merge_branch_name, git_obj.fork_name)
+    if push_details[0] != 0:
+        print(push_details[1])
+        return push_details
+
+    return (0, None)
+
+def create_pr(git_obj, merge_branch_name, work_item_id, username):
+    """
+    Create a pull request using the provided GitRepo object.
+
+    :param git_obj: The GitRepo object to interact with the repository.
+    :param merge_branch_name: The name of the branch where the merge was performed.
+    :param work_item_id: The work item ID to reference in the PR description.
+    :param username: The GitHub username owning the downstream fork.
+    :return: A tuple (status_code, message). Returns (0, None) on success.
+    """
+    pr_details = git_obj.create_pull_request(
+        "Automated Merge pr",
+        get_pr_description(work_item_id),
+        git_obj.local_base_branch,
+        f"{username}:{merge_branch_name}"
+    )
+    if pr_details[0] != 0:
+        print(pr_details[1])
+        return pr_details
+
+    return (0, None)
+
+def get_pr_description(work_item_id):
```

Comment:
If this file is in utils, then it should be usable by everyone, let's not have it know about upstream merge PR description. Could you move this to upstream_merge_and_test and pass the whole PR description here?